src/push/monitor.ts

```diff
@@ -148,9 +151,17 @@ export async function createLightweightMonitors(
   options: PushOptions
 ) {
   const lwFiles = new Set<string>();
+  // Filter monitor files based on the provided pattern
+  const pattern = options.pattern
+    ? new RegExp(options.pattern, 'i')
+    : /.(yml|yaml)$/;
   const ignore = /(node_modules|.github)/;
   await totalist(workDir, (rel, abs) => {
-    if (!ignore.test(rel) && /.(yml|yaml)$/.test(rel)) {
+    if (
+      !ignore.test(rel) &&
+      pattern.test(rel) &&
+      ALLOWED_LW_EXTENSIONS.includes(extname(abs))
```

Comment:
my question about the insensitive case extension would also applied here since includes is case sensitive